MSAL/src/MSALPublicClientApplication.m

```diff
@@ -196,123 +211,106 @@ - (id)initWithClientId:(NSString *)clientId
-    _expirationBuffer = 300;  //in seconds, ensures catching of clock differences between the server and the device
     
     return self;
 }
 
+- (id)initWithClientId:(NSString *)clientId
+         keychainGroup:(NSString *)keychainGroup
+             authority:(MSALAuthority *)authority
+           redirectUri:(NSString *)redirectUri
+                 error:(NSError * __autoreleasing *)error
+{
+    if (!(self = [super init]))
+    {
+        return nil;
+    }
+    MSALPublicClientApplicationConfig *config = [[MSALPublicClientApplicationConfig alloc] initWithClientId:clientId redirectUri:redirectUri authority:authority];
+    
+#if TARGET_OS_IPHONE
+    config.cacheConfig.keychainSharingGroup = keychainGroup ?: [[NSBundle mainBundle] bundleIdentifier];
+#endif
+    
+    return [self initWithConfiguration:config error:error];
+}
+
 #pragma mark - Accounts
 
 - (NSArray <MSALAccount *> *)allAccounts:(NSError * __autoreleasing *)error
 {
     MSALAccountsProvider *request = [[MSALAccountsProvider alloc] initWithTokenCache:self.tokenCache
-                                                                          clientId:self.clientId];
+                                                                            clientId:self.configuration.clientId];
     NSError *msidError = nil;
     NSArray *accounts = [request allAccounts:&msidError];
     if (error) *error = [MSALErrorConverter msalErrorFromMsidError:msidError];
-    
+
     return accounts;
 }
 
 - (MSALAccount *)accountForHomeAccountId:(NSString *)homeAccountId
                                    error:(NSError * __autoreleasing *)error
 {
     MSALAccountsProvider *request = [[MSALAccountsProvider alloc] initWithTokenCache:self.tokenCache
-                                                                          clientId:self.clientId];
+                                                                            clientId:self.configuration.clientId];
     NSError *msidError = nil;
     MSALAccount *account = [request accountForHomeAccountId:homeAccountId error:&msidError];
-    
+
     if (error) *error = [MSALErrorConverter msalErrorFromMsidError:msidError];
-    
+
     return account;
 }
 
 - (MSALAccount *)accountForUsername:(NSString *)username
                               error:(NSError * __autoreleasing *)error
 {
     MSALAccountsProvider *request = [[MSALAccountsProvider alloc] initWithTokenCache:self.tokenCache
-                                                                          clientId:self.clientId];
+                                                                            clientId:self.configuration.clientId];
     NSError *msidError = nil;
     MSALAccount *account = [request accountForUsername:username error:&msidError];
-    
+
     if (error) *error = [MSALErrorConverter msalErrorFromMsidError:msidError];
-    
+
     return account;
 }
 
+
 - (void)allAccountsFilteredByAuthority:(MSALAccountsCompletionBlock)completionBlock
```

Comment:
can we deprecate this one?

Reply:
sure - "Use other synchronous account retrieval API instead.